### DISCUSSION TITLE
List special folders

### DISCUSSION THREAD

**Discussion by Orgbrat** [microsoftgraph/msgraph-sdk-dotnet#1944] [Q&A]
So I have a MAUI app that Reads and Writes a file to the users OneDrive App folder for my application. This all works fine using Microsoft.Graph 4.53.0 but when I update to version 5.12.0  everything breaks and will not compile for missing objects in 5.12.0.

An example is the code line  "graphClient.Me.Drive.Special.AppRoot.Children" as the object "Special" no longer exists.

I did find the doc [Microsoft Graph .NET SDK v5 changelog and upgrade guide](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/upgrade-to-v5.md) but that doc does not help and does not talk about the users special App folder for his installed application to read and write files to.

So does anyone know if there is some migration document, web page,  web site , or sample code some where that explains how to update and fix your code when migrating from 4.53.0 to 5.12.0?

**Comment by andrueastman** [comment]
Due to the paths missing in the current metadata used in generating the SDK, this can achieved as

```cs
var myDrive = await _graphClient
    .Me
    .Drive
    .GetAsync();

var appFolder = await _graphClient
    .Drives[myDrive.Id]
    .Special["approot"]
    .GetAsync();

var fileStream = await _graphClient
    .Drives[myDrive.Id]
    .Items[appFolder.Id]
    .Children["salaries.db"]
    .Content
    .GetAsync();
```

**Reply by Orgbrat** [reply]
@andrueastman  Thank you very much for your quick answer.  I need to access the file from within an app folder "MyAppName"  that is inside the OneDrive Apps folder.  How do I modify the above code to get the file in that location? 

Your help is much appreciated...

**Reply by andrueastman** [reply]
Hey @Orgbrat 

You should be able to change the final statement to use the path relative to the folder to come up with something close to this.

```cs
            var fileStream = await graphClient
                .Drives[myDrive.Id]
                .Items[appFolder.Id]
                .ItemWithPath("MyAppName/file.db")
                .Content
                .GetAsync();
```
